api/generate_proto.sh

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# !/bin/sh
+
+# This file generates API sources from the protocol buffers defined in this
+# directory using Bazel, then copies them back into the source tree so they can
+# be checked-in.
+
+set -e
+set -x
+BAZEL_BINDIR=$(bazel info bazel-bin)
```

Comment:
Why do we use Bazel here? Can we use standard Go module to build?

If we use Go toolchain, there is no need to autogen Bazel build targets.

Reply:
We need `bazel` to build the proto files. This ensures our build is hermetic, and not dependent on any developer's machine or the tools installed therein. Instead, all protos always get built using `bazel`s toolchain, which will be pinned to the versions specified in the `WORKSPACE` file. 

Note that `go build` for building the server continues to work as before though.